packaging/irods.service

```diff
@@ -0,0 +1,20 @@
+[Unit]
+Description=iRODS Open Source Data Management Software
+After=network.target auditd.service
```

Comment:
that's good to know about being ignored - that was an open question.

agreed on the local-fs

Reply:
[sssd](https://sssd.io/) is an NSS/PAM caching daemon, usefull in HPC world when you dont want to be constantly looking up uid/guid over the network (it has its own challenges but I'll leave that rant for another time). In our setup we need it before because otherwise the systemd service can't lookup the iRODS system user and group id's and it errors (ask me how I know), so having it set as `After` wouldn't work for us.

We've fettled with ours a bit since this, so I'll go and look it up later, but the real reason I'm telling you all this is we are probbaly not alone in needing to set some dependencies before (or indeed after) the iRODS service itself, and it would be nice to have a way to include these on initial configuration, if possible. I'm not sure how other services handle this (override files?)

Just dug our latest one out, we needed some modifications to go from Red Hat to Ubuntu 18;
Note this is actually an ansible deployed Jinja2 template
```
[Unit]
Description=iRODS 4.X Open Source Data Management Software for user {{ irods_system_user }}
Documentation=<our wiki>
Wants=sssd.service
After=local-fs.target network.target sssd.service

[Service]
WorkingDirectory=/var/lib/irods
User={{ irods_system_user }} 
Group={{ irods_system_group }}
Environment="PATH=/bin:/usr/bin:/usr/sbin:/var/lib/irods:/var/lib/irods/iRODS"
{% if ansible_distribution == 'Ubuntu' -%}
Environment="LD_LIBRARY_PATH=/usr/lib/oracle/11.1/client64/lib"
{% endif -%}
ExecStart=/bin/bash -c "irodsctl start"
ExecReload=/usr/bin/touch /etc/irods/core.re
ExecStop=/bin/bash -c "irodsctl stop"
KillMode=control-group
Restart=on-failure
RestartPreventExitStatus=255
Type=forking

[Install]
WantedBy=multi-user.target
```

Notes from my colleagues in our git repo on this that might be helpful;

>    Starting with iRODS 4.2, iRODS starts two server processes. If the Main
>    PID fails but the other process doesn't, then systemd tries to start
>    up the unit again (while warning about the still-running process)
>    which fails because iRODS won't start when there's still and
>    irodsServer process extant.
>    
>    Setting KillMode to control-group means in this situation that systemd
>    will send a SIGTERM to the other process then after 90s send it a
>    SIGKILL; and then can successfully start iRODS up again.
 
ref #2774    
 
Also, relating to the above change from target to service;

>     Targets are groups of services - we want to depend on sssd.service not
>    .target. Also, declare a Wants relationship as well as After - if
>    you're tring to start irods you want sssd to be started as well if
>    possible.

You'll notice that, as (currently) an Oracle shop, we have hardcoded the path to the package we had to make out of the RPM (urgh). This is because bionic doesn't set LD_LIBRARY_PATH in the environment of a service (but RHEL7 does), so it needed to be added _somewhere_ so iRODS could find it later (I think it was OK on install and setup, but fails on service restart, but my memory may be incorrect). You may need to consider/make available a similar setup for people using the oracle database plugin (or tell us what were doing wrong!).

The path environment was there to support both 4.1.x and 4.2.x. This may be of a less concern to you if you're distributing it with the package!

Finally, and again relating to the path, we found that (and I have to quote my colleagues magnificent commit message verbatim;

> systemd doesn't actually honour the PATH, so we need to hand off responsibility to 
> something with fewer stoats, like /bin/bash

My what an exiting adventure SystemD is!